string actionmode_desc
string actionmode_name
int8 actionmode_flag